modules/workers/instancepools.tf

```diff
@@ -61,3 +61,63 @@ resource "oci_core_instance_pool" "workers" {
     }
   }
 }
+
+resource "oci_core_instance_pool" "autoscaled_workers" {
+  # Create an OCI Instance Pool resource for each enabled entry of the worker_pools map with that mode.
+  for_each                  = { for key, value in local.enabled_instance_pools: key => value if lookup(value, "ignore_size", false) == true }
```

Comment:
Might play it safe and wrap the lookup with `tobool` here and line 7 - values found in maps can be cast to string on you and unexpectedly fail equality checks.